libraries/chain/abi_serializer.cpp

```diff
@@ -218,17 +218,23 @@ namespace eosio { namespace chain {
       return ends_with(type, "[]");
    }
 
-   bool abi_serializer::is_szarray(const string_view& type)const {
+   std::optional<fc::unsigned_int> abi_serializer::is_szarray(const string_view& type)const {
```

Comment:
I think we need a test that demonstrates unmatched `std::array` size does not work and generates a meaningful error.

Reply:
done.